src/form/form-password-input.tsx

```diff
@@ -1,19 +1,54 @@
 import { useField } from 'formik';
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 
+import { Divider } from '../components/divider/divider';
+import { PasswordStrengthIndicator } from '../components/password-strength-indicator/password-strength-indicator';
 import { StyledPasswordInput } from '../components/styled-password-input/styled-password-input';
 import { StyledPasswordInputProps } from '../components/styled-password-input/styled-password-input.props';
+import { formatSize } from '../styles/format-size';
 import { hasError } from '../utils/has-error';
+import { isDefined } from '../utils/is-defined';
 import { ErrorMessage } from './error-message/error-message';
 
+const MIN_PASSWORD_LENGTH = 8;
+const uppercaseLowercaseMixtureRegx = /(?=.*[a-z])(?=.*[A-Z])/;
+const lettersNumbersMixtureRegx = /(?=.*\d)(?=.*[A-Za-z])/;
+const specialCharacterRegx = /[!@#$%^&*()_+\-=\]{};':"\\|,.<>?]/;
+
+export interface PasswordValidation {
+  minChar: boolean;
+  cases: boolean;
+  number: boolean;
+  specialChar: boolean;
+}
+
 interface Props extends Pick<StyledPasswordInputProps, 'testID'> {
   name: string;
+  isNewPassword?: boolean;
```

Comment:
strange naming, its used not everywhere?

Reply:
yes. Only for when user enters new wallet password